test/IntegrationTestCommon/Factories/WebApplicationFactoryBase.cs

```diff
@@ -177,17 +179,16 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
 
         builder.ConfigureTestServices(services =>
         {
-            var dbContextOptions = services.First(sd => sd.ServiceType == typeof(DbContextOptions<DatabaseContext>));
+            var dbContextOptions =
+                services.First(sd => sd.ServiceType == typeof(DbContextOptions<DatabaseContext>));
             services.Remove(dbContextOptions);
-            services.AddScoped(services =>
-            {
-                return new DbContextOptionsBuilder<DatabaseContext>()
-                    .UseSqlite(SqliteConnection)
-                    .UseApplicationServiceProvider(services)
-                    .Options;
-            });
 
-            MigrateDbContext<DatabaseContext>(services);
+            // Add database to the service collection
+            TestDatabase.AddDatabase(services);
+            if (HandleDbDisposal)
+            {
+                TestDatabase.Migrate(services);
+            }
```

Comment:
`HandleDbDisposal` no longer seems like an accurate name if we're also using it to trigger migrations.